adapter/internal/discovery/xds/server.go

```diff
@@ -277,16 +277,23 @@ func UpdateAPI(vHost string, apiProject model.ProjectAPI, environments []string)
 		return nil, err
 	}
 
-	if apiYaml.APIType == constants.HTTP || apiYaml.APIType == constants.WEBHOOK {
-		err = mgwSwagger.GetMgwSwagger(apiProject.OpenAPIJsn)
-
-		// Set the following in case they were overridden by the above line
-		mgwSwagger.SetID(apiYaml.ID)
-		mgwSwagger.SetName(apiYaml.Name)
-		mgwSwagger.SetVersion(apiYaml.Version)
-		if err != nil {
-			return nil, err
-		}
+	err = apiProject.APIYaml.ValidateAPIType()
```

Comment:
+1. Moving this into the method `ValidateMandatoryFields()`, that gets called at `NewAPIYaml, which does the validation of API name, version and context.

Reply:
The method `PopulateFromAPIYaml()` exists in mgwSwagger because the method populates it. `ValidateAPIType()` gets called at multiple other places. But we can make sure that it is called for all API types and both CC modes if we call it here too. I guess we can move it above the method `PopulateFromAPIYaml()`

Reply:
Fixed with https://github.com/wso2/product-microgateway/pull/2648/commits/67159a07b1ce16a2abe1e8ab796c5b9ec65cbe0e